src/core/lib/iomgr/error.h

```diff
@@ -127,7 +127,11 @@ typedef enum {
 #define GRPC_ERROR_OOM ((grpc_error*)2)
 #define GRPC_ERROR_RESERVED_2 ((grpc_error*)3)
 #define GRPC_ERROR_CANCELLED ((grpc_error*)4)
-#define GRPC_ERROR_SPECIAL_MAX GRPC_ERROR_CANCELLED
+#define GRPC_ERROR_RESERVED_3 ((grpc_error*)5)
+// GRPC_ERROR_EOS is used only by transports for send ops that failed because
+// the stream was closed for writes.
+#define GRPC_ERROR_EOS ((grpc_error*)6)
```

Comment:
Is this actually safe?  I thought we could only use 1-4, because the other bits can all be used by valid pointers.  But I don't remember the details of this -- please double-check with @nicolasnoble.

Do we actually need another special error here?  Why not just create a new error in this case with an appropriate attribute?

Reply:
well, if 4 (0100) is ok, 6 (0110) should also be ok. I am using a special error here because this case needs to detect that there are no other child errors and no other attributes. A special error seemed like the way to do it since it also has the added advantage. (If there are other attributes, it would mean that there were other reasons for the failure and it might make sense to cancel the call in that case.)

Reply:
What @yashykt said :)

lower 4 bits = values from 0 to 15.